src/main/java/school/lemon/changerequest/java/introduction/hw2/Task5.java

```diff
@@ -2,6 +2,12 @@
 
 public class Task5 {
     public static boolean commondEnd(int[] a, int[] b) {
+        int lengthA = a.length;
+        int lengthB = b.length;
+
+        if (lengthA >= 1 && lengthB >= 1) {
+            return a[0] == b[0] || a[a.length - 1] == b[b.length - 1] || a[0] == b[b.length - 1] || a[a.length - 1] == b[0];
+        }
         return false;
     }
```

Comment:
And again. It do pass all tests, but it's again to complicated.
Why do you need this `if` clause ? Also there is no need for `lengthA` and `lengthB` vars (or you should reuse them everywhere).
By the task description: you'll 100% have arrays with length>=1.